### PR TITLE
Update to qupath-0.2.0-m5

### PR DESCRIPTION
**Pull request by red1408**

### CODE REVIEW

File: src/main/java/com/quantumsoft/qupathcloud/pyramid/Pyramid.java [hunk 1]
```diff
@@ -33,13 +35,13 @@
...
-     * Instantiates a new Pyramid.
-     *
-     * @param instances the instances
-     * @throws QuPathCloudException if an exception occurs
-     */
-    public Pyramid(List<Instance> instances) throws QuPathCloudException {
+  /**
+   * Instantiates a new Pyramid.
+   *
+   * @param instances the instances
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/pyramid/Pyramid.java:41]: Change the comment to add the new parameter

File: src/main/java/com/quantumsoft/qupathcloud/pyramid/LoadPyramidFileCallable.java [hunk 2]
```diff
@@ -29,20 +29,22 @@
...
   private Path filePath;
+  private boolean metadataOnly;
 
   /**
    * Instantiates a new Load pyramid file callable.
    *
    * @param filePath the file path
    */
-  public LoadPyramidFileCallable(Path filePath) {
+  public LoadPyramidFileCallable(Path filePath, boolean metadataOnly) {
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/pyramid/LoadPyramidFileCallable.java:41]: Add java doc for new param

File: src/main/java/com/quantumsoft/qupathcloud/imageserver/CloudImageServerBuilder.java [hunk 3]
```diff
@@ -55,8 +52,15 @@ public float supportLevel(URI uri, FileFormatInfo.ImageCheckType info, Class<?>
   }
 
   @Override
-  public ImageServer<BufferedImage> buildServer(URI uri) throws QuPathCloudException {
-    return new CloudImageServer(uri, cloudDao, Repository.INSTANCE.getDicomStore());
+  public UriImageSupport<BufferedImage> checkImageSupport(URI uri, String... args) {
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/imageserver/CloudImageServerBuilder.java:61]: Add Doc stating why this new method is added and what the params mean.

**Review by poojavenkatram — COMMENTED** [review]
Thanks for the changes just a few small comments on Docs.

Best
Pooja

File: src/main/java/com/quantumsoft/qupathcloud/pyramid/Pyramid.java (hunk 1 above)
Comment (red1408) [reply src/main/java/com/quantumsoft/qupathcloud/pyramid/Pyramid.java:41]: Ok

File: src/main/java/com/quantumsoft/qupathcloud/pyramid/LoadPyramidFileCallable.java (hunk 2 above)
Comment (red1408) [reply src/main/java/com/quantumsoft/qupathcloud/pyramid/LoadPyramidFileCallable.java:41]: Ok

File: src/main/java/com/quantumsoft/qupathcloud/imageserver/CloudImageServerBuilder.java (hunk 3 above)
Comment (red1408) [reply src/main/java/com/quantumsoft/qupathcloud/imageserver/CloudImageServerBuilder.java:61]: Ok.
API changes, previously supportLevel(...) itself was interface method, now it's wrapped into UriImageSupport class.

File: src/main/java/com/quantumsoft/qupathcloud/pyramid/Pyramid.java [hunk 4]
```diff
@@ -65,81 +70,97 @@ public Pyramid(List<Instance> instances) throws QuPathCloudException {
...
-    /**
-     * Get downsamples double [ ].
-     *
-     * @return the double [ ]
-     */
-    public double[] getDownsamples() {
+  /**
+   * Get downsamples double [ ].
+   *
+   * @return the double [ ]
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/pyramid/Pyramid.java:77]: The double[] is just the type I think it's better to explain what downsamples are...

**Review by poojavenkatram — COMMENTED** [review]
Thanks for the changes, looks good just one small comment.

File: src/main/java/com/quantumsoft/qupathcloud/pyramid/Pyramid.java (hunk 4 above)
Comment (red1408) [reply src/main/java/com/quantumsoft/qupathcloud/pyramid/Pyramid.java:77]: Ok

**Review by poojavenkatram — APPROVED** [review]
LGTM

**Pull request merged**